src/python/pants/backend/python/util_rules/dists.py

```diff
@@ -248,7 +252,7 @@ async def run_pep517_build(request: DistBuildRequest, python_setup: PythonSetup)
             if line.startswith(f"{dist_type}: "):
                 paths[dist_type] = line[len(dist_type) + 2 :].strip()
     # Note that output_digest paths are relative to the working_directory.
-    output_digest = await Get(Digest, RemovePrefix(result.output_digest, dist_dir))
+    output_digest = await Get(Digest, RemovePrefix(result.output_digest, output_path))
```

Comment:
I don't think this will do what you want? You're stripping the entire output_path, so the final result will, I think, go under `dist/` again.  Or did you test this and it works and I am misreading this code?

Reply:
I'm not sure of my interpretation of the existing code, but I thought since it was stripping out `"dist"` to get the location of the distribution files, we'd now have to strip the full `output_path`.

I haven't been able to test this at all, so I don't know whether it works.